editor/plugins/polygon_2d_editor_plugin.cpp

```diff
@@ -989,9 +990,32 @@ void Polygon2DEditor::_uv_draw() {
 	mtx.columns[2] = -uv_draw_ofs;
 	mtx.scale_basis(Vector2(uv_draw_zoom, uv_draw_zoom));
 
-	RS::get_singleton()->canvas_item_add_set_transform(uv_edit_draw->get_canvas_item(), mtx);
-	uv_edit_draw->draw_texture(base_tex, Point2());
-	RS::get_singleton()->canvas_item_add_set_transform(uv_edit_draw->get_canvas_item(), Transform2D());
+	// Draw texture as a background if editing uvs or no uv mapping exist.
+	if (uv_edit_mode[0]->is_pressed() || uv_mode == UV_MODE_CREATE || node->get_polygon().size() == 0 || node->get_uv().size() != node->get_polygon().size()) {
+		RS::get_singleton()->canvas_item_add_set_transform(uv_edit_draw->get_canvas_item(), mtx);
+		uv_edit_draw->draw_texture(base_tex, Point2());
+		RS::get_singleton()->canvas_item_add_set_transform(uv_edit_draw->get_canvas_item(), Transform2D());
+		preview_polygon->hide();
+	} else {
+		preview_polygon->set_transform(mtx);
+		preview_polygon->set_texture(node->get_texture());
+		preview_polygon->set_texture_offset(node->get_texture_offset());
+		preview_polygon->set_texture_rotation(node->get_texture_rotation());
+		preview_polygon->set_texture_scale(node->get_texture_scale());
+		preview_polygon->set_texture_filter(node->get_texture_filter_in_tree());
+		preview_polygon->set_texture_repeat(node->get_texture_repeat_in_tree());
+		preview_polygon->set_polygon(node->get_polygon());
+		preview_polygon->set_uv(node->get_uv());
+		preview_polygon->set_invert(node->get_invert());
+		preview_polygon->set_invert_border(node->get_invert_border());
```

Comment:
This code comes with a risk that adding a new Polygon2D property may cause the editor to behave wrong when someone forgets to add it here too. Maybe it's possible to use `get_property_list()` to do it automatically? Or at least there should be some comment probably.

Reply:
I do see the concern and initially I tested using `duplicate()` to get the properties, but the problem is that it needs to be decided on case by case basis. Do you think it would be more future proof if we iterate over all properties and make a list of exclusions?
Note that I skip properties `offset`, `skeleton`, `bones` and `vertex_colors`, also all Node2D, CanvasItem and Node properties. If some property gets added to Node2D for example, it could be harder to notice a bug.

I think adding a comment is a good idea. Something like this?
```
// NOTE: Add here if Polygon2D gets new properties which should be reflected on preview.
```

Reply:
Ah, I didn't consider properties of base classes.

Something like this would be more consistent with other similar comments:
```
// Keep in sync with newly added Polygon2D properties (when relevant).
```